lib/improver/cli/spot_extract.py

```diff
@@ -68,6 +68,14 @@ def main(argv=None):
     parser.add_argument("diagnostic_filepath", metavar="DIAGNOSTIC_FILEPATH",
                         help="Path to a NetCDF file containing the diagnostic "
                              "data to be extracted.")
+    parser.add_argument("temperature_lapse_rate_filepath",
+                        metavar="LAPSE_RATE_FILEPATH", nargs='?',
```

Comment:
May be a silly question, but why `nargs='?'`?  Why not just an optional argument?

Reply:
Ah I see this simplifies the call (as you don't need the `--temperature_lapse_rate_filepath` before the name).

Reply:
Not only does it simplify it, this was the major change to enable the taskfilelist generator to provide the required filename for a given cycle/validity time.